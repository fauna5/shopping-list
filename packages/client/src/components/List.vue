<template>
  <div>
    <ul class="shopping-list">
      <!-- TODO handle empty state -->
      <li v-for="item in this.list" :key="item.title">
        <list-item
          :title="item.title"
          :complete="item.complete"
          :id="item.id"
        />
      </li>
    </ul>
  </div>
</template>

<script>
import { mapState } from 'vuex';
import ListItem from './ListItem';

export default {
  name: 'list',
  components: {
    ListItem,
  },
  computed: {
    ...mapState({
      list: state => state.list,
    }),
  },
};
</script>

<style scoped>
.shopping-list {
  list-style-type: none;
  padding: 0px;
  margin: 0px 5px;
}
</style>