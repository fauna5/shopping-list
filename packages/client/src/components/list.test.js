import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import createStore from '../store/store';

import List from './List';

const INIT_LIST = [
  {
    id: 0,
    title: 'eggs',
    complete: false,
  },
  {
    id: 1,
    title: 'milk',
    complete: true,
  },
  {
    id: 2,
    title: 'bread',
    complete: false,
  },
];


const localVue = createLocalVue();
localVue.use(Vuex);

describe('List', () => {
  it('renders', () => {
    const wrapper = shallowMount(List, {
      store: createStore(INIT_LIST),
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
