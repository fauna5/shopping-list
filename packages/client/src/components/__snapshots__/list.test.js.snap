// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders 1`] = `
<div>
  <ul
    class="shopping-list"
  >
    <li>
      <list-item-stub
        id="0"
        title="eggs"
      />
    </li>
    <li>
      <list-item-stub
        complete="true"
        id="1"
        title="milk"
      />
    </li>
    <li>
      <list-item-stub
        id="2"
        title="bread"
      />
    </li>
  </ul>
</div>
`;
